mindmeld/components/entity_recognizer.py

```diff
@@ -119,16 +118,19 @@ def fit(self,
         # Load labeled data
         examples, labels = self._get_examples_and_labels(queries)
 
-        # Build entity types set
-        self.entity_types = set()
-        for label in labels:
-            for entity in label:
-                self.entity_types.add(entity.entity.type)
+        if examples:
+            # Build entity types set
+            self.entity_types = set()
+            for label in labels:
+                for entity in label:
+                    self.entity_types.add(entity.entity.type)
+
+            model = create_model(model_config)
+            model.initialize_resources(self._resource_loader, examples, labels)
+            model.fit(examples, labels)
+            self._model = model
+            self.config = ClassifierConfig.from_model_config(self._model.config)
 
```

Comment:
Modified this snippet similar to RoleClassifiers. We don't have to instantiate a model unless there is some training data. Although this will never be the case with latest changes to _nlp.py_ in this PR. The latest changes ensure that we don't call `EntityRecognizer.fit()` when there are zero entity_types in training data.